src/main/java/com/owczarczak/footballers/footballer/FootballerController.java

```diff
@@ -14,62 +13,50 @@
 @RequestMapping("/footballers")
 public class FootballerController {
 
-    //TODO change to service
     @Autowired
-    FootballerRepository repository;
+    FootballerService service;
 
     @GetMapping
-    public String index(Footballer footballer) {
+    public String index(FootballerDto footballer) {
         return "Greetings !";
     }
 
     @GetMapping("/")
-    public List<Footballer> getAllFootballers() {
-        List<Footballer> footballers = repository.findAll();
-        return footballers;
+    public List<FootballerDto> getAllFootballers() {
+        return service.getAllFootballers();
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Optional<Footballer>> getFootballer(@PathVariable int id) {
-        if (!repository.existsById(id)) {
-            return ResponseEntity.notFound().build();
-        }
-        return ResponseEntity.ok(repository.findById(id));
+    public Optional<FootballerDto> getFootballerById(@PathVariable int id) {
+        return service.getFootballerById(id);
     }
 
```

Comment:
Nie zwracaj Optional w kontrolerze.

Jeśli Optional z serwisu jest pusty to zwróć ResponseEntity z kodem HTTP/404, jeśli coś jest to zwróć HTTP/200 z DTO.

Reply:
Resolved